src/parser.ml

```diff
@@ -332,13 +332,100 @@ let setext_heading s =
   | Some _ | None ->
       raise Fail
 
+let is_punct = function
+  | '!' | '"' | '#' | '$' | '%' | '&' | '\'' | '(' | ')'
+  | '*' | '+' | ',' | '-' | '.' | '/' | ':' | ';' | '<'
+  | '=' | '>' | '?' | '@' | '[' | '\\' | ']' | '^' | '_'
+  | '`' | '{' | '|' | '}' | '~' -> true
+  | _ -> false
+
+let parse_attributes = function
+  | None -> empty_attributes
+  | Some s ->
+    let attributes = String.split_on_char ' ' s in
+    let f acc s =
+      match s.[0] with
+      | '#' -> {acc with id = Some (String.sub s 1 (String.length s - 1))}
+      | '.' -> {acc with classes = (String.sub s 1 (String.length s - 1))::acc.classes}
+      | _ ->
+        let attr = String.split_on_char '=' s in
+        try
+          {acc with attributes = (List.nth attr 0, List.nth attr 1)::acc.attributes}
+        with
+          | Failure _ -> acc
```

Comment:
Here it is better to use pattern matching on `attr` rather than `List.nth` and catching the exception.

Reply:
Done, and this way to do things would cut the attributes value, if it contained an equal (I only used the first element) so I added a concat